<a href="https://colab.research.google.com/github/raimundosena92/Corso_SenacIA_Inteligencia-Artificial/blob/main/UC4_Aula2_estatistica_media_mediana_moda.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Aula 2 – Média, Mediana, Moda e Média Ponderada

## UC4 – Estatística aplicada à Inteligência Artificial

### Objetivos da aula
- Conceituar com clareza média, mediana, moda e média ponderada
- Entender quando cada medida faz mais sentido
- Comparar as medidas em diferentes contextos
- Resolver muitos exercícios, do básico ao aplicado
- Relacionar essas medidas com análise de dados e Inteligência Artificial

























































## Roteiro da aula

1. Revisão rápida: o que é uma medida de tendência central  
2. Média aritmética simples  
3. Mediana  
4. Moda  
5. Média ponderada  
6. Comparação entre as medidas  
7. Efeito dos valores extremos  
8. Exemplos aplicados  
9. Lista grande de exercícios  
10. Desafio final

## 1. O que são medidas de tendência central?

Medidas de tendência central são valores que tentam representar o "centro" de um conjunto de dados.

Elas ajudam a resumir informações.

Exemplos:
- qual é a nota típica de uma turma?
- qual é o salário típico de um grupo?
- qual tamanho de camiseta aparece com mais frequência?
- qual foi, em média, a quantidade vendida por dia?

As principais medidas de tendência central são:
- **média**
- **mediana**
- **moda**

Nesta aula, também veremos a **média ponderada**, que é muito importante em contextos escolares, indicadores e análise de dados.

Outras médias: Geométrica, Harmônica, Trunca, Ponto Médio e Winsorized.

## 2. Média aritmética simples

A média aritmética simples é a soma de todos os valores dividida pela quantidade de valores.

### Fórmula

$$
\bar{x} = \frac{x_1 + x_2 + x_3 + \dots + x_n}{n}
$$

### Ideia intuitiva
A média tenta "equilibrar" os dados.



### Exemplo 1
Notas: 6, 8 e 10

$$
\text{média} = \frac{6 + 8 + 10}{3} = \frac{24}{3} = 8
$$

Logo, a média é 8.

In [ ]:
#media winsorized
salarios = [2000,2000,5000,20000,100000]


In [ ]:
salarios2 = [2000,2000,5000,20000,20000]

In [ ]:
media01 = sum(salarios)/len(salarios)


In [ ]:
media02 = sum(salarios2)/len(salarios2)

In [ ]:
media01

25800.0

In [ ]:
media02

9800.0

In [ ]:
# Media aritmetica com Python
notas = [6, 8, 10]


In [ ]:
sum(notas)/len(notas)
media = sum(notas)/len(notas)
media

8.0

In [ ]:
# media com numpy
import numpy as np
media_numpy = np.mean(notas)
media_numpy

np.float64(8.0)

In [ ]:
# media com statistics
import statistics
media_stat = statistics.mean(notas)
media_stat




8

### Exemplo 2
Idades: 18, 20, 22, 24, 26

$$
\text{média} = \frac{18+20+22+24+26}{5} = \frac{110}{5} = 22
$$

A média é 22.



In [ ]:
idades = [18, 20, 22, 24, 26]


In [ ]:
# Media com Python
sum(idades)/len(idades)
media = sum(idades)/len(idades)
media

22.0

In [ ]:
# Media com Numpy
import numpy as np
media_numpy = np.mean(idades)
media_numpy

np.float64(22.0)

### Quando a média é útil?
A média é útil quando queremos um valor representativo geral e quando os dados não têm valores extremos muito distorcidos.

### Exercício

Calcule a média aritmética do "EXEMPLO 2" utilizando python, numpy e statistics

In [ ]:
# media idades com numpy
import numpy as np
media_numpy = np.mean(idades)
media_numpy

np.float64(22.0)

In [ ]:
# media com statistics
import statistics
media_stat = statistics.mean(idades)
media_stat

22

## 3. Mediana

A mediana é o valor que fica no meio dos dados **depois que eles são colocados em ordem**.



### Passo a passo
1. Ordenar os dados
2. Verificar a quantidade de elementos
3. Encontrar o valor central

### Caso 1: quantidade ímpar de dados
Exemplo: 3, 5, 8, 10, 12

O valor do meio é 8.  
Logo, a mediana é 8.

### Caso 2: quantidade par de dados
Exemplo: 4, 6, 8, 10

Os dois valores centrais são 6 e 8.  (6+8)/2=7
A mediana será a média entre eles: a medidana=7

$$
\text{mediana} = \frac{6+8}{2}=7
$$

### Fórmula para encontrar a média quando o número de elementos (n) é impar:


$$Md ={\left(\frac{n+1}{2}\right)}$$

### Fórmula para encontrar a média quando o número de elementos (n) é par:

$$Md = \frac{{\left(\frac{n}{2}\right)} + {\left(\frac{n}{2} + 1\right)}}{2}$$


### Exercício:

Crie um código python para calcular a mediana (sem utilizar bibliotecas externas, como numpy, stati, etc)

dica: para ordenar os dados, utilize ``dados ordenados = sorted(dados)``

Utilizando o numpy e statistics:

In [ ]:
# Usando IA
def mediana(lista):
    lista_ordenada = sorted(lista)
    n = len(lista_ordenada)

    if n % 2 == 0:  # n é par
        meio1 = lista_ordenada[n//2 - 1]
        meio2 = lista_ordenada[n//2]
        return (meio1 + meio2) / 2
    else:  # n é ímpar
        return lista_ordenada[n//2]

# Exemplo
dados = [2, 4, 6, 8, 10, 12]
print(mediana(dados))  # saída: 5.0



7.0


In [ ]:
# feito pelo professor - Victor
dados = [1,5,7,2,3,4,1]
dados_ordenados = sorted(dados)
dados_ordenados

[1, 1, 2, 3, 4, 5, 7]

In [ ]:
n  = len(dados_ordenados)
n

7

In [ ]:
dados_ordenados =[1, 1, 4, 5, 7]
if n % 2 != 0:   # se n for impar
  mediana = dados_ordenados[n//2]
else:
  indice2 = n//2
  indice1 = indice2 - 1
  mediana = (dados_ordenados[indice1] + dados_ordenados[indice2])/2
mediana

5

In [ ]:
dados = [1,5,7,2,3,4]

In [ ]:
def calcular_mediana(dados):
  #Ordenando dados e capturando quantidadede de elementos
    dados_ordenada = sorted(dados)
    n = len(dados_ordenados)
    #para num impar
    if n % 2 !=0:
      meio = n//2
      mediana = dados_ordenada[meio]
    #para num par
    else:
      indice2 = n//2
      indice1 = indice2-1
      mediana = (dados_ordenados[indice1] + dados_ordenada[indice2])/2
    return mediana
    mediana

In [ ]:
dados = [1,5,7,2,3,4,1]

In [ ]:
calcular_mediana(dados)

2

In [ ]:
# utilizando numpy
mediana01 = np.median(dados)
mediana01

np.float64(3.0)

In [ ]:
lista2 = [1,5,7,2,3,4,1]

In [ ]:
mediana02 = np.median(lista2)
mediana02

np.float64(3.0)

In [ ]:
# utilizando statistic
mediana_lista01 = statistics.median(dados)
mediana_lista01


3

### Por que a mediana é importante?
A mediana é muito útil quando há valores extremos.

Exemplo:
Salários: 1800, 1900, 2000, 2100, 50000

A média sobe muito por causa do 50000, mas a mediana continua mais próxima da realidade do grupo.

In [ ]:
salarios = [1800, 1900, 2000, 2100, 50000]  # quantidade impar ( o elemento central = 2000)
mediana = np.median(salarios)
mediana


np.float64(2000.0)

In [ ]:
salarios = [1800, 1900, 2000, 2100]  # quantidade par (os elementos centrais 1900 e 2000)
mediana = np.median(salarios)
mediana

np.float64(1950.0)

## 4. Moda

A moda é o valor que mais se repete.

### Exemplos

- Dados: 2, 3, 3, 5, 8  
  Moda = 3

- Dados: azul, azul, verde, vermelho, azul  
  Moda = azul

A moda é a única das medidas de tendência central que pode ser aplicada facilmente também em dados qualitativos.

### Observações importantes
- um conjunto pode ter **uma moda**: unimodal
- pode ter **duas modas**: bimodal
- pode ter **mais de duas modas**: multimodal
- pode **não ter moda**, se nenhum valor repetir

In [ ]:
from collections import Counter
dados = [2, 3, 3, 5, 8]
contagem = Counter(dados)
contagem

Counter({2: 1, 3: 2, 5: 1, 8: 1})

In [ ]:
contagem.most_common(1) # contagem mais comum ou seja que se repete( o nº3 repete 2 vezes)

[(3, 2)]

In [ ]:
dados = [2,3,3,5,8]

In [ ]:
dados2 = [2,3,3,3,5,5,8,5]

In [ ]:
import statistics
moda = statistics.mode(dados)
moda

3

In [ ]:
moda1 = statistics.mode(dados)
moda1

3

In [ ]:
moda2 = statistics.multimode(dados2)
moda2

[3, 5]

### Exemplo de moda em dados qualitativos
Categorias de produto:
- eletrônico
- eletrônico
- acessório
- acessório
- acessório

Moda = acessório

In [ ]:
moda_str = statistics.mode(["eletrônico", "eletrônico", "acessório", "acessório", "acessório"])
moda_str

'acessório'

In [ ]:
dados_str = ["a", "b", "c", "b"]

In [ ]:
moda_str = statistics.mode(dados_str)
moda_str

'b'

## 5. Média ponderada

Na média simples, todos os valores têm o mesmo peso.

Na média ponderada, alguns valores valem mais do que outros.

### Fórmula

$$
\text{média ponderada} = \frac{x_1p_1 + x_2p_2 + x_3p_3 + \dots + x_np_n}{p_1+p_2+p_3+\dots+p_n}
$$

onde:
- \(x\) são os valores
- \(p\) são os pesos

### Exemplo 1: notas com pesos diferentes
Um aluno tirou:
- Prova 1 = 7, peso 2
- Prova 2 = 9, peso 3

$$
\text{média ponderada} = \frac{7 \cdot 2 + 9 \cdot 3}{2+3}
= \frac{14+27}{5}
= \frac{41}{5}
= 8,2
$$

Logo, a média ponderada é 8,2.

In [1]:
notas = [7, 9]
pesos = [2, 3]

media_ponderada = sum(n*p for n, p in zip(notas, pesos)) / sum(pesos)
media_ponderada
# zip
# n= notas
# p=peso

8.2

In [2]:
import numpy as np

notas = [7, 9]
pesos = [2, 3]

media_ponderada = np.average(notas, weights=pesos)
print(media_ponderada)

8.2


In [ ]:
import statistics

# Cria uma lista onde o 7 aparece 2 vezes e o 9 aparece 3 vezes
dados_expandidos = [7, 7, 9, 9, 9]
media = statistics.mean(dados_expandidos)
media
#Aqui tem uma observação importante

8.2

### Exemplo 2: média ponderada em compras
Um produto foi vendido assim:
- 10 unidades a R\$ 5
- 30 unidades a R\$ 8

O preço médio ponderado por quantidade é:

$$
\frac{10 \cdot 5 + 30 \cdot 8}{10+30}
=
\frac{50+240}{40}
=
\frac{290}{40}
=
7,25
$$

O valor 7,25 representa melhor o preço médio real das vendas do que uma média simples entre 5 e 8.

In [3]:
precos = [5, 8]
quantidades = [10, 30]

sum(p*q for p, q in zip(precos, quantidades)) / sum(quantidades)

7.25

## 6. Comparando as medidas

| Medida | Ideia principal | Ponto forte | Limitação |
|---|---|---|---|
| Média | equilíbrio dos dados | usa todos os valores | sensível a extremos |
| Mediana | valor central | resiste a outliers | não usa a intensidade completa dos dados |
| Moda | valor mais frequente | útil em categorias | pode não existir ou haver várias |
| Média ponderada | média com pesos diferentes | mais realista em muitos contextos | exige pesos corretos |

### Regra prática
- use **média** quando os dados forem relativamente equilibrados
- use **mediana** quando houver valores extremos
- use **moda** para descobrir o valor mais comum
- use **média ponderada** quando os valores tiverem importâncias diferentes

## 7. Efeito dos valores extremos

Vamos comparar dois conjuntos:

### Conjunto A
10, 12, 13, 14, 16

### Conjunto B
10, 12, 13, 14, 100

Ambos têm vários valores parecidos, mas o segundo tem um valor extremo.

In [4]:
conjunto_a = [10, 12, 13, 14, 16]
conjunto_b = [10, 12, 13, 14, 100]

resultado = {
    "media_A": sum(conjunto_a)/len(conjunto_a),
    "mediana_A": np.median(conjunto_a),
    "media_B": sum(conjunto_b)/len(conjunto_b),
    "mediana_B": np.median(conjunto_b),
}
resultado

{'media_A': 13.0,
 'mediana_A': np.float64(13.0),
 'media_B': 29.8,
 'mediana_B': np.float64(13.0)}

### Conclusão
No conjunto B, a média sobe bastante por causa do valor 100, mas a mediana permanece mais estável.

Isso mostra por que a mediana é tão importante em dados com outliers.

## 8. Aplicações em IA e análise de dados

Essas medidas aparecem em várias tarefas:

### Média
- média de notas
- média de vendas
- média de tempo de atendimento
- média de erro de um modelo

### Mediana
- valor típico quando há outliers
- análise de salários
- tempo de resposta de sistemas
- limpeza de dados

### Moda
- categoria mais frequente
- classe mais comum
- produto mais vendido
- valor usado em preenchimento de dados categóricos faltantes

### Média ponderada
- nota final com pesos
- indicadores compostos
- métricas combinadas
- média de preços por volume

# Exercícios resolvidos

## Exercício resolvido 1
Calcule a média de 4, 6, 8, 10

$$
\text{média} = \frac{4+6+8+10}{4} = \frac{28}{4} = 7
$$

In [ ]:
dados = [4, 6, 8, 10]
media = sum(dados)/len(dados)
media

7.0

## Exercício resolvido 2
Calcule a mediana de 2, 5, 7, 9, 12

Os dados já estão em ordem.  
O valor central é 7.  
Logo, a mediana é 7.

In [ ]:
mediana = np.median([2, 5, 7, 9, 12])
mediana

np.float64(7.0)

## Exercício resolvido 3
Calcule a moda de 1, 2, 2, 3, 4, 4, 4, 5

O valor que mais repete é 4.  
Logo, a moda é 4.

## Exercício resolvido 4
Calcule a média ponderada:
- 6 com peso 1
- 8 com peso 2
- 10 com peso 3

$$
\frac{6\cdot1 + 8\cdot2 + 10\cdot3}{1+2+3}
=
\frac{6+16+30}{6}
=
\frac{52}{6}
=
8,67
$$

In [ ]:
valores = [6, 8, 10]
pesos = [1, 2, 3]

media_ponderada = sum(v*p for v, p in zip(valores, pesos)) / sum(pesos)
media_ponderada

8.666666666666666

# Lista de exercícios

## Parte A – Conceituação

### Exercício 1
Explique, com suas palavras:
- o que é média ->é a soma dos dados dividido pelo qtdade dados
- o que é mediana -> é o valor central dos dados(meio)
- o que é moda -> é o dado que mais se repete
- o que é média ponderada -> é a média de dados com pesos diferentes.

### Exercício 2
Qual medida de tendência central costuma ser mais afetada por valores extremos? Explique. A media, pois fogem muito da realidade dos dados.

### Exercício 3
Em que situação a moda é mais útil do que a média?
qdo preciso saber o valor do dado mais comum, o que mais se repete.

### Exercício 4
Qual a diferença entre média simples e média ponderada?
a media simples é calculado sobre dados de mesmo peso e na media ponderado temos que considerar os pesos dos dados que sao diferentes.

## Parte B – Cálculo direto

### Exercício 5
Calcule a média:
- 5, 7, 9
media = (5+7+9)/3
media =21/3 = 7

### Exercício 6
Calcule a média:
- 12, 15, 18, 21
media=(-12+15+18+21)/4
media=42/4 = 4.5

### Exercício 7
Calcule a mediana:
- 3, 5, 8, 10, 12
mediana= 8

### Exercício 8
Calcule a mediana:
- 4, 6, 8, 10
mediana=(6+8)/2
mediana= 7

### Exercício 9
Calcule a moda:
- 2, 2, 3, 4, 5
moda= 2: 2 (unimodal)

### Exercício 10
Calcule a moda:
- 1, 3, 3, 4, 4, 6
moda= 3: 2, 4: 2 (bimodal)

### Exercício 11
Calcule média, mediana e moda:
- 2, 4, 4, 5, 7
media=(2+4+4+5+7)/5 -> media=22/5 = 4.4
mediana = 4
moda= 4: 2(unimodal)

### Exercício 12
Calcule média, mediana e moda:
- 10, 10, 12, 15, 20
media=(-10+10+12+15+20)/5 -> media=47/5 =9.4
mediana = 12
moda = nao existe pois, nenhum numero se repete.

In [6]:
### Exercício 5
dados = [5, 7, 9]
media = sum(dados)/len(dados)
media


7.0

In [11]:
### Exercício 6
dados = [12, 15, 18, 21]
media = sum(dados)/len(dados)
media


16.5

In [14]:
# Exercicio 7
dados_ordenados =[3, 5, 8, 10, 12]
n = len(dados_ordenados)
if n % 2 != 0:   # se n for impar
  mediana = dados_ordenados[n//2]
else:
  indice2 = n//2
  indice1 = indice2 - 1
  mediana = (dados_ordenados[indice1] + dados_ordenados[indice2])/2
mediana

8

In [15]:
# Exercicio 9
from collections import Counter
dados = [2, 2, 3, 4, 5]
contagem = Counter(dados)
contagem

Counter({2: 2, 3: 1, 4: 1, 5: 1})

In [16]:
contagem.most_common(1) # contagem mais comum ou seja que se repete( o nº2 repete 2 vezes)

[(2, 2)]

In [20]:
# Exercicio 10
from collections import Counter
dados = [1, 3, 3, 4, 4 , 6]
contagem = Counter(dados)
contagem

Counter({1: 1, 3: 2, 4: 2, 6: 1})

In [22]:
contagem.most_common(2) # contagem mais comum ou seja que se repete( o nº3 e n repete 2 vezes)

[(3, 2), (4, 2)]

In [24]:
import statistics


In [25]:
moda = statistics.multimode(dados)
moda

[3, 4]

In [ ]:
### Exercício 11
Calcule média, mediana e moda:
- 2, 4, 4, 5, 7
media=(2+4+4+5+7)/5 -> media=22/5 = 4.4
mediana = 4
moda= 4: 2(unimodal)


In [26]:
# calculo da media
dados = [2, 4, 4, 5, 7]
media = sum(dados)/len(dados)
media

4.4

In [27]:
# Calculo da mediana
dados_ordenados =[2, 4, 4, 5, 7]
n = len(dados_ordenados)
if n % 2 != 0:   # se n for impar
  mediana = dados_ordenados[n//2]
else:
  indice2 = n//2
  indice1 = indice2 - 1
  mediana = (dados_ordenados[indice1] + dados_ordenados[indice2])/2
mediana

4

In [28]:
  # Calculo da moda
  from collections import Counter
  dados = [2, 4, 4, 5, 7]
  contagem = Counter(dados)
  contagem




Counter({2: 1, 4: 2, 5: 1, 7: 1})

In [29]:
contagem.most_common(1) # contagem mais comum ou seja que se repete( o nº3 e n repete 2 vezes)

[(4, 2)]

In [ ]:
### Exercício 12
Calcule média, mediana e moda:
- 10, 10, 12, 15, 20
media=(-10+10+12+15+20)/5 -> media=47/5 =9.4
mediana = 12
moda = nao existe pois, nenhum numero se repete.

In [30]:
# Media
dados = [-10, 10, 12, 15, 20]
media = sum(dados)/len(dados)
media

9.4

In [31]:
# Calculo da mediana
dados_ordenados =[-10, 10, 12, 15, 20]
n = len(dados_ordenados)
if n % 2 != 0:   # se n for impar
  mediana = dados_ordenados[n//2]
else:
  indice2 = n//2
  indice1 = indice2 - 1
  mediana = (dados_ordenados[indice1] + dados_ordenados[indice2])/2
mediana

12

In [35]:
# Calculo da moda
from collections import Counter
dados = [-10, 10, 12, 15, 20]
contagem = Counter(dados)
contagem
# Nao existe moda

Counter({-10: 1, 10: 1, 12: 1, 15: 1, 20: 1})

In [ ]:
### Exercício 13
Calcule a média ponderada:
- 7 com peso 2
- 9 com peso 3
mp=[(7*2)+(9*3)]/(2+3)
mp=(14+27)/5
mp=41/5 -> mp=8.2

In [36]:
precos = [7, 9]
quantidades = [2, 3]

sum(p*q for p, q in zip(precos, quantidades)) / sum(quantidades)

8.2

In [ ]:
### Exercício 14
Calcule a média ponderada:
- 5 com peso 1
- 6 com peso 2
- 8 com peso 2
mp=[(5*1)+(6*2)+(8*2)]/(1+2+2)
mp=(5+12+16)/5
mp=33/5 -> mp=6.6

In [37]:
precos = [5, 6, 8]
quantidades = [1, 2, 2]

sum(p*q for p, q in zip(precos, quantidades)) / sum(quantidades)

6.6

In [ ]:
### Exercício 15
Um aluno teve as seguintes notas:
- trabalho = 8, peso 2
- prova = 6, peso 5
- projeto = 9, peso 3

In [38]:
precos = [8, 6, 9]
quantidades = [2, 5, 3]

sum(p*q for p, q in zip(precos, quantidades)) / sum(quantidades)

7.3

In [ ]:
### Exercício 16
Um produto foi vendido assim:
- 20 unidades a /R$ 4
- 10 unidades a /R$ 6
- 30 unidades a /R$ 5

In [39]:
precos = [4, 6, 5]
quantidades = [20, 10, 30]

sum(p*q for p, q in zip(precos, quantidades)) / sum(quantidades)

4.833333333333333

## Parte C – Média ponderada

### Exercício 13
Calcule a média ponderada:
- 7 com peso 2
- 9 com peso 3
mp=[(7*2)+(9*3)]/(2+3)
mp=(14+27)/5
mp=41/5 -> mp=8.2

### Exercício 14
Calcule a média ponderada:
- 5 com peso 1
- 6 com peso 2
- 8 com peso 2
mp=[(5*1)+(6*2)+(8*2)]/(1+2+2)
mp=(5+12+16)/5
mp=33/5 -> mp=6.6

### Exercício 15
Um aluno teve as seguintes notas:
- trabalho = 8, peso 2
- prova = 6, peso 5
- projeto = 9, peso 3

Calcule a média final.
mp=[(8*2)+(6*5)+(9*3)]/(2+5+3)
mp=[16+30+27]/10
mp=73/10 -> mp=7.3

### Exercício 16
Um produto foi vendido assim:
- 20 unidades a /R$ 4
- 10 unidades a /R$ 6
- 30 unidades a /R$ 5

Calcule o preço médio ponderado.
mp=[(20*4)+(10*6)+(30*5)]/(4+6+5)
mp=[80+60+150]/15
mp=290/15
Preço medio ponderado

## Parte D – Interpretação

### Exercício 17
Observe os conjuntos:
- A = 8, 9, 10, 10, 11
- B = 8, 9, 10, 10, 100

Calcule a média e a mediana de cada um.  
mediaA=(8+9+10+10+11)/5 -> mediaA=9.6
mediaB=(8+9+10+10+100)/5 -> mediaB=27.4
medianaA=10
medianaB=10
Depois, diga qual medida representa melhor cada conjunto. A mediana representa melhor a tendencia central pois desconsidera o valor discrepante

### Exercício 18
Uma loja registrou as categorias mais vendidas:
- camiseta
- camiseta
- tênis
- calça
- camiseta
- tênis

Qual é a moda?  (camiseta: 3)
Por que a média não faria sentido aqui? A media é usada para dados numericos. A unica medida de tendencia central que mede dados qualitativos é a Moda.

### Exercício 19
Explique por que a mediana pode ser melhor que a média em salários. Porque a mediana desconsidera os valores extremos tanto para baixo quanto para cima o que aproxima mais da medida central.

### Exercício 20
Explique por que a média ponderada é mais adequada para calcular nota final quando provas têm pesos diferentes. Porque a media ponderada considera os pesos de cada dado.

## Parte E – Aplicação com Python

Execute as células abaixo e complete as respostas.

In [ ]:
# Base 1
notas_turma = [6, 7, 8, 8, 9, 10, 5, 8]

media = sum(notas_turma) / len(notas_turma)
mediana = np.median(notas_turma)
moda = Counter(notas_turma).most_common(1)[0][0]

media, mediana, moda

(7.625, np.float64(8.0), 8)

### Exercício 21
Anote os valores de média, mediana e moda da turma acima.
media=7.625
mediana= 8.0
moda=(8: 3)

### Exercício 22
Qual medida parece representar melhor essa turma? Explique. A media representa com mais precisao.

In [ ]:
# Base 2
salarios = [1800, 2000, 2200, 2500, 3000, 50000]

media = sum(salarios) / len(salarios)
mediana = np.median(salarios)

media, mediana

(10250.0, np.float64(2350.0))

### Exercício 23
Compare média e mediana dos salários.A media está muito longe da tendencia central pois considera o valor discrepante, enquanto a media fica mais proxima da medida central pois desconsider valor discrepante.

### Exercício 24
Qual delas representa melhor o "salário típico" desse grupo? Mediana

In [ ]:
# Base 3
valores = [7, 9, 6]
pesos = [2, 5, 3]

media_pond = sum(v*p for v, p in zip(valores, pesos)) / sum(pesos)
media_pond

7.7

### Exercício 25
Interprete o valor encontrado acima. Para dados com pesos diferente a media ponderada é a melhor indicada pois, é a que mais se aproxima do centro.

### Exercício 26
Se todos os pesos fossem iguais, o resultado mudaria? Verifique. Sim, o resultado puxaria a media para baixo pois, a maior nota foi na matéria com o maior peso.
media=(7+9+6)/3 = 7,333

In [ ]:
# Compare média simples e média ponderada
valores = [7, 9, 6]
media_simples = sum(valores) / len(valores)

pesos = [2, 5, 3]
media_ponderada = sum(v*p for v, p in zip(valores, pesos)) / sum(pesos)

media_simples, media_ponderada

(7.333333333333333, 7.7)

# Desafio final

Uma escola quer resumir o desempenho de uma turma.

As notas foram:
- 4, 5, 5, 6, 6, 6, 7, 8, 20

## Tarefas
1. Calcule a média:
media=(4+5+5+6+6+6+7+8+20)/9 -> media=67/9 -> media=7.444

2. Calcule a mediana
mediana=6

3. Calcule a moda
moda=(6: 3.0) (5: 2.0) bimodal

4. Diga qual medida representa melhor a turma
A medidana pois fica mais proximo da tendencia cental

5. Explique por que uma delas pode ser enganosa nesse caso
A media distorce devido a nao desconsiderar os valores discrepantes(outliers)

Depois, responda:

> Se você fosse construir um sistema para analisar automaticamente notas de turmas, qual medida usaria com mais cuidado: média, mediana ou moda? Justifique. Media pois, se na amostra tiver dados discrepantes irá distorcer o resultado.

# Fechamento

Nesta aula, estudamos:
- média
- mediana
- moda
- média ponderada

Também vimos:
- quando usar cada medida
- como valores extremos afetam a média
- por que a mediana pode ser mais adequada em certos contextos
- como a média ponderada aparece em notas, vendas e indicadores

Na próxima etapa, essas medidas servirão de base para interpretação de dados e comparação entre grupos.